File: rust/src/models/slot_pricing.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct SlotPricing {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub max_slots: Option<i32>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub min_slots: Option<i32>,

    pub unit_rate: String,
}

impl SlotPricing {
    pub fn new(unit_rate: String) -> Self {
        Self {
            max_slots: None,
            min_slots: None,
            unit_rate,
        }
    }
}

File: typescript/src/models/slotPricing.ts
// this file is @generated

export interface SlotPricing {
  maxSlots?: number | null;

  minSlots?: number | null;

  unitRate: string;
}

export const SlotPricingSerializer = {
  _fromJsonObject(object: any): SlotPricing {
    return {
      maxSlots: object["max_slots"],
      minSlots: object["min_slots"],
      unitRate: object["unit_rate"],
    };
  },

  _toJsonObject(self: SlotPricing): any {
    return {
      max_slots: self.maxSlots,
      min_slots: self.minSlots,
      unit_rate: self.unitRate,
    };
  },
};
